app/main/views/suppliers.py

```diff
@@ -147,9 +148,74 @@ def import_supplier(supplier_id):
 
         supplier.contact_information.append(contact_information)
 
-    db.session.add(supplier)
+        db.session.add(supplier)
+
+    try:
+        db.session.commit()
+    except IntegrityError as e:
+        db.session.rollback()
+        abort(400, "Database Error: {0}".format(e))
+
+    return jsonify(suppliers=supplier.serialize()), 201
+
+
+@main.route('/suppliers', methods=['POST'])
+def create_supplier():
+    supplier_data = get_json_from_request()
+
+    json_has_required_keys(
+        supplier_data,
+        ['suppliers']
+    )
+
+    supplier_data = supplier_data['suppliers']
+    supplier_data = drop_foreign_fields(supplier_data, ['links'])
+
+    json_has_required_keys(
+        supplier_data,
+        ['name', 'contactInformation']
+    )
+
+    contact_informations_data = [
+        drop_foreign_fields(contact_data, ['links'])
+        for contact_data in supplier_data['contactInformation']
+        ]
+
+    supplier_data['contactInformation'] = contact_informations_data
+
+    for contact_information_data in contact_informations_data:
+        json_has_required_keys(
+            contact_information_data,
+            ['contactName', 'email', 'postcode']
+        )
+
+    validate_new_supplier_json_or_400(supplier_data)
+
+    supplier_data = drop_foreign_fields(
+        supplier_data,
+        ['contactInformation']
+    )
+
+    supplier = Supplier()
+    supplier.update_from_json(supplier_data)
+
+    for contact_information_data in contact_informations_data:
```

Comment:
This has quite a different page flow to the edit pages, no chance of reuse.

Was also thinking that making a supplier in the first instance should require all of the details, which includes min contact details. One API call to make one valid supplier.

Updates though are not the same thing.

Could be done as two, depends if we think that suppliers should or must have a contact.